src/actions/channels.js

```diff
@@ -666,15 +667,11 @@ export function deleteChannel(channelId: string): ActionFunc {
         }
 
         const entities = getState().entities;
-        const {channels, currentChannelId} = entities.channels;
+        const {currentChannelId} = entities.channels;
         if (channelId === currentChannelId && !viewArchivedChannels) {
-            const channel = Object.keys(channels).filter((key) => channels[key].name === General.DEFAULT_CHANNEL);
-            let defaultChannelId = '';
-            if (channel.length) {
-                defaultChannelId = channel[0];
-            }
+            const channel = getChannelByName(state, getRedirectChannelNameForTeam(state, getCurrentTeamId(state)));
```

Comment:
Not specific to this PR, but I wonder if there's a utility to compose selectors like this. It seems like two common ways to use selectors are either to combine the results of many selectors which can be done using `createSelector` or to chain them together like this. It would be nice if you didn't have to pass `state` into them repeatedly.

Reply:
we easily create a function that behave like createSelector but only for returning the result. something like:
```js
function runSelector(...selectors) {
  return (state, options) => {
    let result = options;
    for (const selector of selectors) {
      result = selector(state, result)
    }
    return result
  }
}
let channel = runSelector(
  getCurrentTeamId,
  getRedirectChannelNameForTeam,
  getChannelByName,
)(state)
```

Reply:
But I'm 0/5 about adding it, I don't know if its going to be more clear or not.